test/units.sh

```diff
@@ -1,28 +1,58 @@
 #!/bin/bash
 
-go get github.com/weaveworks/weave/testing/cover
+set -e
 
-if [ -n "$COVERDIR" ] ; then
-	coverdir="$COVERDIR"
-else
-	coverdir=$(mktemp -d coverage.XXXXXXXXXX)
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+SLOW=${SLOW-}
+GO_TEST_ARGS="-tags netgo -cpu 4"
+if [ -n "$SLOW" -o "$1" = "-slow" ]; then
+    GO_TEST_ARGS="$GO_TEST_ARGS -race -covermode=atomic"
+
+    if [ -n "$COVERDIR" ] ; then
+        coverdir="$COVERDIR"
+    else
+        coverdir=$(mktemp -d coverage.XXXXXXXXXX)
+    fi
+
+    mkdir -p $coverdir
 fi
 
-mkdir -p $coverdir
 fail=0
 
-for dir in $(find . -type f -name '*_test.go' | xargs -n1 dirname | grep -v prog | sort -u); do
-	go get -t -tags netgo $dir
-	output=$(mktemp $coverdir/unit.XXXXXXXXXX)
-	if ! go test -cpu 4 -tags netgo -covermode=count -coverprofile=$output $dir ; then
-		fail=1
-	fi
+TESTDIRS=$(find . -type f -name '*_test.go' | xargs -n1 dirname | grep -v prog | sort -u)
```

Comment:
`TESTDIRS=$(find . -type f -name '*_test.go' -exec dirname {} \; | grep -v prog | sort -u)` is slightly more efficient